src/install_pypi/plan/validation.rs

```diff
@@ -248,9 +253,9 @@ pub(crate) fn need_reinstall(
                             // Check the repository base url with the locked url
                             let installed_repository_url =
                                 RepositoryUrl::new(installed_git_url.url.repository());
-                            if locked_git_url.url.repository()
-                                != &installed_repository_url.into_url()
-                            {
+                            let locked_repository_url =
+                                RepositoryUrl::new(locked_git_url.url.repository());
+                            if locked_repository_url != installed_repository_url {
```

Comment:
This would now errorneously do a comparison with fragments, so the test failed, tried to get it to work here.